File: backend/src/main/java/objects/Assignment.java
package objects;

public class Assignment {
    private boolean active=true;
    private int MaxScore;
    private String title;
    private assingmentType type;
    private int deadline; //todo better use Date class for deadline. here just for easy use

    public Assignment(int MaxScore, String title, assingmentType type, int deadline) {
        this.MaxScore = MaxScore;
        this.title = title;
        this.type = type;
        this.deadline = deadline;
    }

    public void setActive(boolean active) {
        this.active = active;
    }

    public void setDeadline(int deadline) {
        this.deadline = deadline;
    }

    public static assingmentType type(String type){
        if(type.toLowerCase().equals("project")){
            return assingmentType.PROJECT;
        } else{
            return assingmentType.EXERCISE;
        }
    }

    public int getMaxScore() {
        return MaxScore;
    }

    public assingmentType getType() {
        return type;
    }

    public String getTitle() {
        return title;
    }
}

enum assingmentType{
    EXERCISE, PROJECT
}

File: frontend/lib/Assignment.dart
class Assignment{
  bool _active = true;
  late int _MaxScore;
  late String _title;
  late AssignmentType _assignmentType;
  late DateTime _deadline;

  Assignment(this._MaxScore, this._title, this._assignmentType, this._deadline);

  int get MaxScore => _MaxScore;
  String get title => _title;
  AssignmentType get assignmentType => _assignmentType;
  DateTime get deadline => _deadline;
  bool get active => _active;

  set active(bool active){
    this._active = active;
  }
  set MaxScore(int MaxScore){
    this._MaxScore = MaxScore;
  }
  set title(String title){
    this._title = title;
  }
  set assignment(AssignmentType assignmentType){
    this._assignmentType = assignmentType;
  }
  set deadLine(DateTime deadLine){
    this._deadline = deadLine;
  }

}
enum AssignmentType{
  EXERCISE, PROJECT
}
